src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinWord = '';
+    const arrCopy = [...this];
+
+    for (let i = 0; i < arrCopy.length; i++) {
+      if (i !== 0) {
+        joinWord += separator;
+      }
+
+      if (arrCopy[i] === undefined || arrCopy[i] === null) {
```

Comment:
Instead of checking `===`, better check `!==` and do only `joinWord += arrCopy[i];`